src/logworker.cpp

```diff
@@ -111,7 +113,7 @@ namespace g3 {
    }
 
    void LogWorker::addWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
-      auto bg_addsink_call = [this, sink] {_impl._sinks.push_back(sink);};
+      auto bg_addsink_call = [this, &sink] {_impl._sinks.push_back(sink);};
```

Comment:
I see what you are doing here but I strongly disagree with it.  Please provide a valid reason that can be proven.  The lambda will not and cannot according to C++ spec, as far as I understand it, stay alive past this function call.